# Copyright (c) 2017-2018 Kent R. Spillner <kspillner@acm.org>
#
# Permission to use, copy, modify, and distribute this software for any
# purpose with or without fee is hereby granted, provided that the above
# copyright notice and this permission notice appear in all copies.
#
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

Add patch to silence OCaml compiler unsafe string errors

--- PKGBUILD.orig	2018-02-23 16:30:14.000000000 -0600
+++ PKGBUILD	2018-02-23 16:32:05.000000000 -0600
@@ -119,6 +119,8 @@
 	'grub.conf'
 	"${pkgbase}.conf"
 	'tmpfiles.conf'
+
+	'ocaml-unsafe-string.patch'
 )
 
 if [ "${_build_stubdom}" = true ] ; then
@@ -161,6 +163,8 @@
 	'3f0af16958c3e057b9baa5afc47050d9adf7dd553274dd97ae4f35938fefb568'
 	'50a9b7fd19e8beb1dea09755f07318f36be0b7ec53d3c9e74f3266a63e682c0c'
 	'80227f8daa62a49c08fb7ffcc7de8cabbd0645396e46c04f7caaa71b04f446f0'
+
+	'1b08e60c38b4a591d1cdb152d9d8ec3f6ca17c8a32058346ab1354ac10424e2d'
 )
 
 
@@ -261,6 +265,8 @@
 	# Copied from Gentoo's xen 4.8.1-r2 ebuild
 	find . \( -name 'Makefile*' -o -name '*.mk' -o -name 'common.make' \) -exec sed -i 's/ *-Werror */ /' '{}' ';'
 	sed -i 's/, "-Werror"//' 'tools/python/setup.py'
+
+	patch -Np1 -i "${srcdir}/ocaml-unsafe-string.patch"
 }
 
 build() {
